src/permutation.jl

```diff
@@ -0,0 +1,65 @@
+export PermutationTest
+
+function ptstats(x,y,f)
+    xy = vcat(x,y)
+    rx = 1 : length(x)
+    ry = length(xy)-length(y)+1 : length(xy)
```

Comment:
Looks like you're constructing the concatenation of `x` and `y`, but only using it to compute the length. This will be inefficient when `x` and/or `y` are large.

`length(vcat(x, y))` should be `length(x) + length(y)`, right? Then `length(vcat(x, y)) - length(y) + 1 == length(y) + length(x) - length(y) + 1`, which simplifies to `length(x) + 1`. So you can write this as
```julia
nx = length(x)
rx = 1:nx
ry = nx+1:nx+length(y)
```